pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,39 @@
+package ipsec
+
+import (
+	"strconv"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedState     string
+	recordedLegacyAPI string
+)
+
+func init() {
+	ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+		Name: "cluster_network_ipsec_state",
+		Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+	}, []string{"mode", "is_legacy_api"})
+	legacyregistry.MustRegister(ipsecStateGauge)
```

Comment:
I am confused - how will the metrics be consumed from CNO? 
We dont expose any endpoint for Prometheus to scrape this metric AFAIK.

Reply:
It will be consumed from the telemetry module - the Prometheus query "openshift:cluster_network_ipsec_state:sum" will be gathered by the telemetry agent

Reply:
I mean, how will prometheus actually get the metrics info - i dont think we expose a server for prometheus agent to scrape the metrics. 
Did you see the metrics being exposed in OCP console with this PR?

Reply:
Yep - 
![image](https://github.com/openshift/cluster-network-operator/assets/41079547/c8da4f6d-8e26-4c92-b723-566336775135)

Reply:
I was wrong - its serving on 9104. Thanks!